packages/desktop/components/popup/popups/ReceiveAddressPopup.svelte

```diff
@@ -4,44 +4,58 @@
     import { generateAndStoreEvmAddressForAccounts, pollL2BalanceForAccount } from '@core/layer-2/actions'
     import { activeProfile, activeProfileId } from '@core/profile/stores'
-    import { checkActiveProfileAuth } from '@core/profile/actions'
+    import { checkActiveProfileAuthAsync } from '@core/profile/actions'
     import { LedgerAppName } from '@core/ledger'
     import PopupTemplate from '../PopupTemplate.svelte'
+    import { handleError } from '@core/error/handlers'
+    import { IAccountState } from '@core/account'
 
-    export let selectedNetworkId: NetworkId = $network.getMetadata().id
+    let selectedNetworkId: NetworkId | undefined = getActiveNetworkId()
     $: selectedNetworkId, updateNetworkNameAndAddress()
 
-    let networkName: string
-    let receiveAddress: string
+    let networkName: string | undefined
+    let receiveAddress: string | undefined
     function updateNetworkNameAndAddress(): void {
-        if (isStardustNetwork(selectedNetworkId)) {
-            networkName = $network.getMetadata().name
-            receiveAddress = $selectedAccount.depositAddress
-        } else if (isEvmChain(selectedNetworkId)) {
-            const { id, name, coinType } = $network.getChain(selectedNetworkId)?.getConfiguration() ?? {}
+        const account = $selectedAccount as IAccountState
+
+        if (selectedNetworkId && isStardustNetwork(selectedNetworkId)) {
+            networkName = $network?.getMetadata().name
+            receiveAddress = account.depositAddress
+        } else if (selectedNetworkId && isEvmChain(selectedNetworkId)) {
+            const chainConfig = $network?.getChain(selectedNetworkId)?.getConfiguration()
+            if (!chainConfig) return
+
+            const { name, coinType } = chainConfig
             networkName = name
-            receiveAddress = $selectedAccount.evmAddresses?.[coinType]
+            receiveAddress = account.evmAddresses?.[coinType]
             if (!receiveAddress) {
-                void checkActiveProfileAuth(
-                    async () => {
-                        await generateAndStoreEvmAddressForAccounts($activeProfile.type, coinType, $selectedAccount)
-                        pollL2BalanceForAccount($activeProfileId as string, $selectedAccount)
-                        networkName = name
-                        receiveAddress = $selectedAccount.evmAddresses?.[coinType]
-                    },
-                    { ledger: true, stronghold: true, props: { selectedNetworkId: id } },
-                    LedgerAppName.Ethereum
-                )
+                generateAddress(account, coinType)
             }
         } else {
             networkName = undefined
             receiveAddress = undefined
         }
     }
 
+    async function generateAddress(account: IAccountState, coinType: number): Promise<void> {
+        try {
+            await checkActiveProfileAuthAsync(LedgerAppName.Ethereum)
+        } catch {
+            return
+        }
```

Comment:
I think something needs to be changed with the function, because I have the problem in multiple places: